src/functions-and-arrays.js

```diff
@@ -1,41 +1,135 @@
 // Iteration #1: Find the maximum
-function maxOfTwoNumbers() {}
+function maxOfTwoNumbers(number1, number2) {
+let result;
+if (number1 > number2) {
+ result = number1
+} else {
+ result = number2
+}
+return result;
+}
+const maxNumber = maxOfTwoNumbers (3,8);
+console.log(maxNumber);
 
 
 
 // Iteration #2: Find longest word
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
 
-function findLongestWord() {}
-
+function findLongestWord(listword) {
+if (listword.length === 0) {
+  return null;
+}
+let longestWord;
+ for (let i = 0; i < listword.length; i++) {
+    if (longestWord === undefined){
+   longestWord = listword[i]
+  } else if (listword[i].length > longestWord.length) {
+    longestWord = listword[i]
+  }
+ }
+return longestWord;
+}
+console.log(findLongestWord(words));
 
 
 // Iteration #3: Calculate the sum
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
-function sumNumbers() {}
-
+function sumNumbers(listnumber) {
+  let numbers = 0;
+  for (let i = 0; i < listnumber.length; i++) {
+    numbers = listnumber[i] + numbers
+  }
+  return numbers;
+}
+ console.log(sumNumbers(numbers));
 
 
 // Iteration #3.1 Bonus:
-function sum() {}
-
+const mixedArr = [6, 12, 'miami', 1, true, 'barca', '200', 'lisboa', 8, 10];
+// should return: 57
+function sum(listall) {
+  let elements = 0;
+  for (let i = 0; i < listall.length; i++) {
+    if (typeof listall[i] === "string") {
+    word = listall[i]
+    elements = word.length + elements
+  } else if (typeof listall[i] === "number") {
+    number = listall[i]
+    elements = number + elements
+  } else if (typeof listall[i] === "boolean") {
+    boolean = listall[i]
+    elements = boolean + elements
+  } else if (typeof listall[i] === "array" || "object") {
```

Comment:
Bien resuelto! pero un detalle, sabes cual es el typeof de los arrays? (ahora viene cuando odiaras un poco js porque esta mal hecho xD) el typeof de array es object, esto en cuanto expliquemos los prototipos y objetos lo entederas un poco mas jejeje pero bien ese OR.

```suggestion
  } else if (typeof listall[i] === "object") {
```